rfcs/0192-fr-copy-receipt-performance.md

```diff
@@ -0,0 +1,218 @@
+We then count all of those CopyReceipts and return that as the count.
+
+# Guide-level explanation
+
+For the PosCreator they are no implementation changes needed, the CopyReceipts still work like they did before.
+
+## New local queues
+
+New Queues don't need to take anything into account.
+They will work out of the box.
+
+## Updating local queues
+
+The new table will be created on start up after updating the queue.
+
+The new `ftJournalFRCopyPayload` table will be populated with the `CopyPayload` (The deserialized JWT) data of each CopyReceipt from the `ftJournalFR` table.
+
+## SignatureCloud
+
+The SignatureCloud will be updated as well.
+The new version will not automatically be used since this would mean a recertification of the PosCreator.
+Instead we will time the switch to the new version accordingly with the PosCreators.
+
+# Reference-level explanation
+
+We will add a migration which adds the new table `ftJournalFRCopyPayload`.
+
+This Table will contain the following columns:
+
+| Name                    | Data Type  |
+|-------------------------|------------|
+| QueueId                 | `Guid`     |
+| CashBoxIdentification   | `string`   |
+| Siret                   | `string`   |
+| ReceiptId               | `string`   |
+| ReceiptMoment           | `DateTime` |
+| QueueItemId             | `Guid`     |
+| CopiedReceiptReference  | `string`   |
+| CertificateSerialNumber | `string`   |
+
+This table contains one row for each CopyReceipt and contains the deserialized JWT of the CopyReceipt taken from the `ftJournalFR` table.
+
+To get the count of copies of one receipt we can simply count the rows with the same `CopiedReceiptReference`.
+
+When a new CopyReceipt is processed we count the rows where the `CopiedReceiptReference` equals the `cbPreviousReceiptReference` and return that as the count.
+Then we insert the JWT payload of the new CopyReceipt into the `ftJournalFRCopyPayload` table.
+
+The `GetCountOfExistingCopiesAsync` method looks like this:
+```cs
+private Task<int> GetCountOfExistingCopiesAsync(string cbPreviousReceiptReference)
+{
+  return _journalFRCopyPayloadRepository.GetCountOfCopiesAsync(cbPreviousReceiptReference);
+}
+```
+
+The `GetCountOfCopiesAsync` method of the SQLite `JournalFRCopyPayloadRepository` looks like this:
+```cs
+public async Task<int> GetCountOfCopiesAsync(string cbPreviousReceiptReference) {
+    var query = "SELECT count(*) FROM ftJournalFRCopyPayload WHERE CopiedReceiptReference = @cbPreviousReceiptReference";
+    return await DbConnection.QuerySingleAsync<int>(query, new { cbPreviousReceiptReference });
+}
+```
+
+## Initialization
+
+The new table will be created with the usual migration process.
+
+On start up of the Queue (in the 1.2 `workerFR.cs`) we check if the table contains any entries and if not we will populate it with the old data from the `ftJournalFR` table.
```

Comment:
I guess it would also check if the queue is started and especially has already processed copy receipts (maybe with the `CNumerator` of the QueueFR?), to not execute this multiple times whenever a Queue is started that doesn't contain copy receipts